### PR TITLE
librbd: restore diff-iterate include_parent functionality in fast-diff mode

### PR DESCRIPTION
**Pull request by idryomov**
## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [x] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/librbd/api/DiffIterate.cc [hunk 1]
```diff
@@ -69,7 +69,7 @@ class C_DiffObject : public Context {
       m_image_ctx, io::IMAGE_DISPATCH_LAYER_INTERNAL_START,
       aio_comp, {{m_image_offset, m_image_length}},
       {m_diff_context.from_snap_id, m_diff_context.end_snap_id},
-      (m_diff_context.include_parent ?
+      ((m_diff_context.include_parent && m_diff_context.from_snap_id == 0) ?
         0 : io::LIST_SNAPS_FLAG_DISABLE_LIST_FROM_PARENT) |
```
Comment (trociny) [inline src/librbd/api/DiffIterate.cc:72]: nit: may be it is time to introduce local `flags` variable, calculate its value and pass to `io::ImageDispatchSpec::create_list_snaps` to make the code more readable?

**Review by trociny — APPROVED** [review]
LGTM

File: src/librbd/api/DiffIterate.cc (hunk 1 above)
Comment (idryomov) [reply src/librbd/api/DiffIterate.cc:72]: Agreed.

**Comment by idryomov** [comment]
jenkins retest this please

**Comment by idryomov** [comment]
No related failures:

https://pulpito.ceph.com/dis-2022-01-08_10:51:32-rbd-wip-dis-testing-distro-default-smithi/
https://pulpito.ceph.com/dis-2022-01-10_22:21:45-rbd-wip-dis-testing-distro-default-smithi/

**Pull request merged**